### DISCUSSION TITLE
In-server event emission

### DISCUSSION THREAD

**Discussion by Indeedornot** [socketio/socket.io#4587] [Q&A] [answered]
Hi!
I have searched high and low for this feature, looking through both default docs and api spect. I am holding some data inside socket's data property, and this data can be changed by other server-side sockets on an event and I would want to inform them of the change to update themselves.

Here's the schematic

![image](https://user-images.githubusercontent.com/49447848/211378981-213c319a-ceae-4b91-80bd-1686ff49cb50.png)

In this case it is done for role management in my app since socket contains a copy of a user to check for permissions rather than having to look it up from database every time

**Comment by darrachequesne** [comment, marked as answer]
Hi! You can loop over the `sockets` map:

```js
io.of("/").sockets.forEach((socket) => {
  // ...
});
```

Reference: https://socket.io/docs/v4/server-api/#namespacesockets

Please note that this only works with a single server, with multiple servers you will need to use the `serverSideEmit()` method.

Reference: https://socket.io/docs/v4/server-api/#serverserversideemiteventname-args

**Reply by Indeedornot** [reply]
Apologies for getting back so late.

Your solution did work! Thank you!
(Although it turned out to be spaghetti in my case and I'll have to move to backend as a service xD)

**Reply by qiulang** [reply]
Or you can let all socket clients join a room and when the specific event happens, you broadcast to that room from your socket.
